internal/token.go

```diff
@@ -1236,7 +1236,13 @@ func (z *Tokenizer) readTagAttrExpression() {
 				if next == '/' {
 					panic("Block comments (//) are not allowed inside of expressions")
 				}
-				z.readCommentOrRegExp()
+				// Also stop when we hit a '}' character (end of attribute expression)
+				z.readCommentOrRegExp([]byte{'}'})
+				// If we exit on a '}', ignore the final character here
+				lastChar := z.buf[z.data.End-1 : z.data.End][0]
+				if lastChar == '}' {
+					z.data.End--
+				}
```

Comment:
This checks if the final character was `}`, in which case we need to do `z.data.End--` to back up and re-process the character in this loop.